### PR TITLE
Fix `mapJSHandle` invalid memory address or nil pointer dereference

### PR DESCRIPTION
**Pull request by inancgumus**
## What?

Add missing function arguments check to `JSHandle` mapping.

## Why?

#1543 

## Checklist

- [x] I have performed a self-review of my code
- [ ] I have added tests for my changes
- [x] I have commented on my code, particularly in hard-to-understand areas

## Related PR(s)/Issue(s)

Updates #1543

### CODE REVIEW

**Review by ankur22 — COMMENTED** [review]
Thanks for the fix! I like where it is going.

Two points:

1. Can we look to fix the other `evaluate`/`evaluateHandle` methods on `page` and `frame`? In a new PR if you prefer, but happy to have them as separate commits in this PR 🙂 
2. This does indeed fix the issues when `evaluate()` is used without pageFunc. It however breaks one thing which is that you can no longer pass in a string:

```js
// As a string that is callable
console.log(await jsHandle.evaluate("node => node.innerText"));
// As a function
console.log(await jsHandle.evaluate(node => node.innerText));
```

3. Do we need tests for at least one of these methods (maybe just on `page`?).

**Comment by inancgumus** [comment]
@ankur22, sure thing!

> 1. Can we look to fix the other `evaluate`/`evaluateHandle` methods on `page` and `frame`? In a new PR if you prefer, but happy to have them as separate commits in this PR 🙂 

Sure :) Can you repurpose issue #1543? Maybe remove [this explanation](https://github.com/grafana/xk6-browser/issues/1543#issuecomment-2486143338) from the issue description so that we can understand what the issue entails. Currently, it looks like it's about `mapJSHandle`.

> 2. This does indeed fix the issues when `evaluate()` is used without pageFunc. It however breaks one thing which is that you can no longer pass in a string:

Are you sure that it doesn't work with strings? I've tried this, and it worked:

```js
const n = await handle.evaluate('() => 5 * 42');
```

But, this is a nice catch because now we don't accept non-functions.

> 3. Do we need tests for at least one of these methods (maybe just on `page`?).

Would adding a test to `examples/` be fine?

**Comment by ankur22** [comment]
> Sure :) Can you repurpose issue https://github.com/grafana/xk6-browser/issues/1543? Maybe move https://github.com/grafana/xk6-browser/issues/1543#issuecomment-2486143338 to the issue description so that we can understand what the issue entails. Currently, it looks like it's about mapJSHandle.

Sure, done.

> Are you sure that it doesn't work with strings?

Yeah, this script below no longer works with this branch:

```js
import { browser } from 'k6/browser';

export const options = {
  scenarios: {
    ui: {
      executor: 'shared-iterations',
      options: {
        browser: {
            type: 'chromium',
        },
      },
    },
  },
}

export default async function() {
  const page = await browser.newPage();

  try {
    await page.goto('https://test.k6.io/my_messages.php');

    const jsHandle = await page.waitForSelector('h2');

    // As a string that is callable
    console.log(await jsHandle.evaluate("node => node.innerText"));
    
    // As a function
    console.log(await jsHandle.evaluate(node => node.innerText));
  } finally {
    await page.close();
  }
}
```

> Would adding a test to examples/ be fine?

I think we should have table tests (integration tests) in the `tests` package, unless there's a reason it's not possible to create them? A test in the examples folder would be a good addition too.

File: tests/page_test.go [hunk 1]
```diff
@@ -264,6 +264,11 @@ func TestPageEvaluateMappingError(t *testing.T) {
 			script:  "(6)",
 			wantErr: "Given expression does not evaluate to a function",
 		},
+		{
+			name:    "void",
+			script:  "",
+			wantErr: "evaluate requires a page function",
+		},
```
Comment (ankur22) [inline tests/page_test.go:271]: Perfect 👍 I was mainly looking for this.

Would be good to get a test to make sure that this works as i demonstrated in the script:

    // As a string that is callable
    console.log(await jsHandle.evaluate("node => node.innerText"));
    // As a function
    console.log(await jsHandle.evaluate(node => node.innerText));

Comment (inancgumus) [reply tests/page_test.go:271]: Don't we already have tests for that?

https://github.com/grafana/xk6-browser/blob/d19534e927bfe19133205d1b16110973f3f86ca4/tests/page_test.go#L228-L239

I've also added a test for `SobekEmptyString`. Lastly, I ran your script, and it works.

**Review by ankur22 — APPROVED** [review]
LGTM 🚀

**Pull request merged**